include/openthread/mesh_diag.h

```diff
@@ -221,6 +227,70 @@ otError otMeshDiagGetNextIp6Address(otMeshDiagIp6AddrIterator *aIterator, otIp6A
  */
 otError otMeshDiagGetNextChildInfo(otMeshDiagChildIterator *aIterator, otMeshDiagChildInfo *aChildInfo);
 
+/**
+ * This type represents information about a child entry from `otMeshDiagQueryChildTable()`
+ *
+ */
+typedef struct otMeshDiagChildEntry
+{
+    bool         mRxOnWhenIdle : 1;    ///< Is rx-on when idle (vs sleepy).
+    bool         mDeviceTypeFtd : 1;   ///< Is device FTD (vs MTD).
+    bool         mFullNetData : 1;     ///< Whether device gets full Network Data (vs stable sub-set).
+    bool         mCslSynchronized : 1; ///< Is CSL capable and CSL synchronized.
+    bool         mSupportsErrRate : 1; ///< Supports tracking of error rates (frame/msg err rate).
+    uint16_t     mRloc16;              ///< RLOC16.
+    otExtAddress mExtAddress;          ///< Extended Address.
+    uint16_t     mVersion;             ///< Version.
+    uint32_t     mTimeout;             ///< Timeout in seconds.
+    uint32_t     mAge;                 ///< Seconds since last heard from the child.
+    uint16_t     mSupervisionInterval; ///< Supervision interval in seconds. Zero to indicate not used.
+    int8_t       mAverageRssi;         ///< Average RSSI.
+    int8_t       mLastRssi;            ///< RSSI of last received frame.
+    uint16_t     mFrameErrorRate;      ///< Frame error rate (0xffff->100%). Zero if not known or supported.
```

Comment:
Yes. Thanks. I agree `Zero if not known` can be confusing. I intended it as if you don't support error rate track, set then to zero, but we can use `mSupportsErrRate` flag to tell if the error rate values are valid, otherwise value should not matter and we should ignore them.

In the next push, I remove this and make it simpler. Also added a comment at the top to explain the error rate a bit more.